jackson-modules/jackson-conversions/src/test/java/com/baeldung/jackson/date/JacksonDateUnitTest.java

```diff
@@ -161,6 +157,29 @@ public void whenSerializingJava8Date_thenCorrect() throws JsonProcessingExceptio
         assertThat(result, containsString("2014-12-20T02:30"));
     }
 
+    @Test
+    public void whenSerializingJava8DateAndReadingValue_thenCorrect() throws IOException {
+        final String stringDate = "\"2014-12-20\"";
+
+        final ObjectMapper mapper = new ObjectMapper();
+        mapper.registerModule(new JavaTimeModule());
+        mapper.disable(SerializationFeature.WRITE_DATES_AS_TIMESTAMPS);
+
+        final String result2 = mapper.readValue(stringDate, LocalDate.class)
+            .toString();
```

Comment:
This is an odd way to test serialization: load it into an object and then do `toString` on that object. Can we not assert the fields of the LocalDate object? At a glance, this just looks like string processing, even though there's a date object in the middle.